packages/safe-web3-lib/src/Web3Adapter.ts

```diff
@@ -53,6 +53,10 @@ class Web3Adapter implements EthAdapter {
     return BigNumber.from(await this.#web3.eth.getBalance(address))
   }
 
+  async getNonce(address: string): Promise<number> {
```

Comment:
Why do we need to proxy this function? If we really do, please also add a second optional param called `blockTag`.

Reply:
What can the `blockTag` be used for?

Reply:
In the app, we're passing `pending` to fetch unfinished txen, because that's how you know the latest nonce.

Reply:
Actually, we should just probably pass `pending` here, like in safe-react, w/o making it a parameter.

Reply:
`getNonce` is not used directly when using the user `nonce` from the `TransactionOptions` type.
It is used in tests and it could also be useful for people playing around with the `nonce` property.

I added an optional argument that allows to specify the `blockTag`, adding it to methods that were not receiving it as well.